frontend/src/components/home/HomePage.tsx

```diff
@@ -136,7 +136,6 @@ export default class HomePage extends React.Component<HomePageProps, HomePageSta
             isLoading: false,
             displayUpdateNotice: false,
           });
-          this.checkUpdate();
```

Comment:
You are right, I thought that in this case `setState()` would behave as an asynchronous method, as usual, but it doesn't, as I just checked. I will revert this change in the next commit.